pkg/assembler/backends/neo4j/pkg.go

```diff
@@ -246,3 +253,210 @@ func (e *nameToVersion) PropertyNames() []string {
+
+			sb.WriteString("MATCH (n:Pkg)-[:PkgHasType]->(type:PkgType)-[:PkgHasNamespace]->(namespace:PkgNamespace)-[:PkgHasName]->(name:PkgName)-[:PkgHasVersion]->(version:PkgVersion)")
+
+			if pkgSpec.Type != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "type", "type", "$pkgType")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "type", "type", "$pkgType")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgType"] = pkgSpec.Type
+			}
+			if pkgSpec.Namespace != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "namespace", "namespace", "$pkgNamespace")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "namespace", "namespace", "$pkgNamespace")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgNamespace"] = pkgSpec.Namespace
+			}
+			if pkgSpec.Name != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "name", "name", "$pkgName")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "name", "name", "$pkgName")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgName"] = pkgSpec.Name
+			}
+			if pkgSpec.Version != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "version", "version", "$pkgVerion")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "version", "version", "$pkgVerion")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgVerion"] = pkgSpec.Version
+			}
+
+			if pkgSpec.Subpath != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "version", "subpath", "$pkgSubpath")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "version", "subpath", "$pkgSubpath")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgSubpath"] = pkgSpec.Subpath
+			}
+
+			if pkgSpec.MatchOnlyEmptyQualifiers != nil && !*pkgSpec.MatchOnlyEmptyQualifiers {
+
+				if len(pkgSpec.Qualifiers) > 0 {
+
+					for _, qualifier := range pkgSpec.Qualifiers {
+						// neo4j does not accept "." in its properties. If the qualifier contains a "." that must
+						// be replaced by an "-"
+						qualifierKey := strings.ReplaceAll(qualifier.Key, ".", "_")
```

Comment:
nit: make helper function in case we need to change this in the future.

Reply:
added helper.